src/System/Random/Internal.hs

```diff
@@ -0,0 +1,1010 @@
+  uniformRM (CWchar b, CWchar t) = fmap CWchar . uniformRM (b, t)
+  {-# INLINE uniformRM #-}
+
+instance Uniform CSigAtomic where
+  uniformM = fmap CSigAtomic . uniformM
+instance UniformRange CSigAtomic where
+  uniformRM (CSigAtomic b, CSigAtomic t) = fmap CSigAtomic . uniformRM (b, t)
+  {-# INLINE uniformRM #-}
+
+instance Uniform CLLong where
+  uniformM = fmap CLLong . uniformM
+instance UniformRange CLLong where
+  uniformRM (CLLong b, CLLong t) = fmap CLLong . uniformRM (b, t)
+  {-# INLINE uniformRM #-}
+
+instance Uniform CULLong where
+  uniformM = fmap CULLong . uniformM
+instance UniformRange CULLong where
+  uniformRM (CULLong b, CULLong t) = fmap CULLong . uniformRM (b, t)
+  {-# INLINE uniformRM #-}
+
+instance Uniform CIntPtr where
+  uniformM                         = fmap CIntPtr . uniformM
+instance UniformRange CIntPtr where
+  uniformRM (CIntPtr b, CIntPtr t) = fmap CIntPtr . uniformRM (b, t)
+  {-# INLINE uniformRM #-}
+
+instance Uniform CUIntPtr where
+  uniformM = fmap CUIntPtr . uniformM
+instance UniformRange CUIntPtr where
+  uniformRM (CUIntPtr b, CUIntPtr t) = fmap CUIntPtr . uniformRM (b, t)
+  {-# INLINE uniformRM #-}
+
+instance Uniform CIntMax where
+  uniformM = fmap CIntMax . uniformM
+instance UniformRange CIntMax where
+  uniformRM (CIntMax b, CIntMax t) = fmap CIntMax . uniformRM (b, t)
+  {-# INLINE uniformRM #-}
+
+instance Uniform CUIntMax where
+  uniformM = fmap CUIntMax . uniformM
+instance UniformRange CUIntMax where
+  uniformRM (CUIntMax b, CUIntMax t) = fmap CUIntMax . uniformRM (b, t)
+  {-# INLINE uniformRM #-}
+
+instance UniformRange CFloat where
+  uniformRM (CFloat l, CFloat h) = fmap CFloat . uniformRM (l, h)
+  {-# INLINE uniformRM #-}
+
+instance UniformRange CDouble where
+  uniformRM (CDouble l, CDouble h) = fmap CDouble . uniformRM (l, h)
+  {-# INLINE uniformRM #-}
+
+
+-- The `chr#` and `ord#` are the prim functions that will be called, regardless of which
+-- way you gonna do the `Char` conversion, so it is better to call them directly and
+-- bypass all the hoops. Also because `intToChar` and `charToInt` are internal functions
+-- and are called on valid character ranges it is impossible to generate an invalid
+-- `Char`, therefore it is totally fine to omit all the unnecessary checks involved in
+-- other paths of conversion.
+word32ToChar :: Word32 -> Char
+word32ToChar (W32# w#) = C# (chr# (word2Int# w#))
+{-# INLINE word32ToChar #-}
+
+charToWord32 :: Char -> Word32
+charToWord32 (C# c#) = W32# (int2Word# (ord# c#))
+{-# INLINE charToWord32 #-}
+
+instance Uniform Char where
+  uniformM g = word32ToChar <$> unbiasedWordMult32 (charToWord32 maxBound) g
+  {-# INLINE uniformM #-}
+instance UniformRange Char where
+  uniformRM (l, h) g =
+    word32ToChar <$> unbiasedWordMult32RM (charToWord32 l, charToWord32 h) g
+  {-# INLINE uniformRM #-}
+
+instance Uniform Bool where
+  uniformM = fmap wordToBool . uniformWord8
+    where wordToBool w = (w .&. 1) /= 0
```

Comment:
Is this faster than `odd`?

Reply:
I would suspect so, but it will be worth trying out